modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/tree/BPlusTree.java

```diff
@@ -2169,6 +2169,20 @@ public List<L> remove(L lower, L upper, int limit) throws IgniteCheckedException
         return Collections.unmodifiableList(rmvOp.removedRows);
     }
 
+    /**
+     * @param lower Lower bound (inclusive).
+     * @param upper Upper bound (inclusive).
+     * @param x Implementation specific argument.
+     * @param limit Limit of processed entries by single call, {@code 0} for no limit.
+     * @return {@code True} if removed at least one row.
+     * @throws IgniteCheckedException If failed.
+     */
+    protected boolean removex(L lower, L upper, Object x, int limit) throws IgniteCheckedException {
```

Comment:
Minor: is this method going to be used from several places? Should it be inlined?

Reply:
This method uses private class and private method, so can't be inlined in other classes.